manim/utils/color/core.py

```diff
@@ -536,10 +600,53 @@ def interpolate(self, other: ManimColor, alpha: float) -> ManimColor:
         ManimColor
             The interpolated ManimColor
         """
-        return ManimColor(
-            self._internal_value * (1 - alpha) + other._internal_value * alpha
+        return self._construct_from_space(
+            self._internal_space * (1 - alpha) + other._internal_space * alpha
         )
 
+    def opacity(self, opacity: float) -> Self:
+        """Creates a new ManimColor with the given opacity and the same color value as before
+
+        Parameters
+        ----------
+        opacity : float
+            The new opacity value to be used
+
+        Returns
+        -------
+        ManimColor
+            The new ManimColor with the same color value but the new opacity
+        """
+        tmp = self._internal_space.copy()
+        tmp[-1] = opacity
+        return self._construct_from_space(tmp)
+
+    __INTO = TypeVar("__INTO", bound="ManimColor")
```

Comment:
I would put this outside the class (type vars defined as class attributes aren't documented), and maybe rename it to `ColorT` (I'm not a fan of `__INTO`)

Reply:
I also just realized that something like this already exists as `ManimColorT`, that seems like the better `TypeVar` to use.